### ISSUE TITLE
Snapping: Resizing blocks in relation to the page

### ISSUE THREAD

**Issue by swissspidy** [ampproject/amp-wp#2992]
See #2875 for details.

Example: if the block is 200px from the left side then resizing the right size should allow snapping to 200px from the right side as well.

**Comment by swissspidy** [comment]
Screencast: https://cloudup.com/cqBfxMlZFGF

Shows functionality from  this, #2993, #2994, and #2995.

**Comment by swissspidy** [comment]
**Testing Instructions**

1. Create new story
1. Move the pre-existing block around on the page
1. For each of the block's sides (left, right, top, bottom) and centers (vertical and horizontal), verify that red snapping guidelines are shown in the following scenarios:  
    1. When resizing the block to the sides of the page (left, right, top, bottom).
    1. When resizing the block to the center (vertical and horizontal) of the page.

Basically, snapping guidelines should be shown to allow easily aligning blocks to the sides and center of the page while doing the resising. No actual _snapping_ is done.

Next, verify that no guidelines are shown when resizing and at the same time holding down the alt/option key.

**Comment by csossi** [comment]
Verified in QA
